test/common/routes_exist.go

```diff
@@ -0,0 +1,224 @@
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "plugin-registry",
+			isTLS: true,
+		},
+	},
+
+	"middleware-monitoring-operator": []expectedRoute{
+		expectedRoute{
+			Name:  "alertmanager-route",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "grafana-route",
+			isTLS: true,
+		},
+		expectedRoute{
+			Name:  "prometheus-route",
+			isTLS: true,
+		},
+	},
+
+	"rhsso": []expectedRoute{
+		expectedRoute{
+			Name:  "keycloak",
+			isTLS: false,
+		},
+		expectedRoute{
+			Name:  "keycloak-edge",
+			isTLS: true,
+		},
+	},
+
+	"solution-explorer": []expectedRoute{
+		expectedRoute{
+			Name:  "tutorial-web-app",
+			isTLS: true,
+		},
+	},
+
+	"ups": []expectedRoute{
+		expectedRoute{
+			Name:  "ups-unifiedpush-proxy",
+			isTLS: true,
+		},
+	},
+
+	"user-sso": []expectedRoute{
+		expectedRoute{
+			Name:  "keycloak",
+			isTLS: false,
+		},
+		expectedRoute{
+			Name:  "keycloak-edge",
+			isTLS: true,
+		},
+	},
+}
+
+const (
+	intlyNamespacePrefix = "redhat-rhmi-"
+)
+
+// TestIntegreatlyRoutesExist tests that the routes for all the products are created
+func TestIntegreatlyRoutesExist(t *testing.T, ctx *TestingContext) {
+	var err error
+
+	for product, routes := range expectedRoutes {
+		for _, expectedRoute := range routes {
+			err = checkRoute(t, ctx, product, expectedRoute)
+
+			if err != nil {
+				t.Errorf("Failed checking route %v for product %v: %v",
+					expectedRoute.Name, product, err)
+			}
+		}
+	}
+}
+
+func checkRoute(t *testing.T, ctx *TestingContext, product string, expectedRoute expectedRoute) error {
+	var err error
+
+	if expectedRoute.IsGeneratedName {
+		err = checkRouteByGeneratedName(t, ctx, product, expectedRoute)
+	} else {
+		err = checkRouteByName(t, ctx, product, expectedRoute)
+	}
+
+	return err
+}
+
+func checkRouteByName(t *testing.T, ctx *TestingContext, product string, expectedRoute expectedRoute) error {
+	route := &routev1.Route{}
+	err := ctx.Client.Get(goctx.TODO(), k8sclient.ObjectKey{Name: expectedRoute.Name, Namespace: intlyNamespacePrefix + product}, route)
+
+	if err != nil {
+		return err
+	}
+
+	// Check for TLS in the route
+	return checkRouteTLS(expectedRoute.isTLS, route)
+}
+
+func checkRouteByGeneratedName(t *testing.T, ctx *TestingContext, product string, expectedRoute expectedRoute) error {
+	routes := &routev1.RouteList{}
+
+	// Get the routes for the product
+	err := ctx.Client.List(goctx.TODO(), routes, &k8sclient.ListOptions{
+		Namespace: intlyNamespacePrefix + product,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	// Iterate through the routes and look for the one that matches the expected one.
+	// If it's found, check if TLS matches the expected
+	for _, route := range routes.Items {
+		// Skip routes that don't point to services
+		if route.Spec.To.Kind != "Service" {
+			continue
+		}
+
+		generatedName := route.GetObjectMeta().GetGenerateName()
+		to := route.Spec.To.Name
+
+		if expectedRoute.Name == generatedName && expectedRoute.ServiceName == to {
```

Comment:
I added the logic to check for the service in order to avoid false positives:
If `expectedRoute.Name` is "keycloak" and there's a route called "keycloak-edge" the `strings.HasPrefix` call would be evaluated to true.
Also there's the case in 3scale:
```go
expectedRoute{
	Name:            "zync-3scale-api-",
	IsGeneratedName: true,
	isTLS:           true,
	ServiceName:     "apicast-staging",
},
expectedRoute{
	Name:            "zync-3scale-api-",
       IsGeneratedName: true,
	isTLS:           true,
	ServiceName:     "apicast-production",
},
```
Where they have the same generated name so the test would pass even if there's only one of these routes available